PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/action/CreateMySQLAction.java

```diff
@@ -0,0 +1,106 @@
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.Project;
+import com.microsoft.azure.management.mysql.v2020_01_01.Server;
+import com.microsoft.azure.toolkit.intellij.mysql.creation.MySQLCreationDialog;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTask;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import com.microsoft.azure.toolkit.lib.mysql.AzureMySQLConfig;
+import com.microsoft.azure.toolkit.lib.mysql.AzureMySQLService;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.azuretools.ijidea.actions.AzureSignInAction;
+import com.microsoft.azuretools.utils.AzureUIRefreshCore;
+import com.microsoft.azuretools.utils.AzureUIRefreshEvent;
+import com.microsoft.intellij.AzurePlugin;
+import com.microsoft.intellij.util.AzureLoginHelper;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.helpers.Name;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureActionEnum;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.mysql.MySQLModule;
+
+import javax.swing.*;
+
+import static com.microsoft.intellij.ui.messages.AzureBundle.message;
+
+@Name(CreateMySQLAction.ACTION_NAME)
+public class CreateMySQLAction extends NodeActionListener {
+
+    protected static final String ACTION_NAME = "Create";
+
+    private final MySQLModule model;
+
+    public CreateMySQLAction(MySQLModule model) {
+        super();
+        this.model = model;
+    }
+
+    @Override
+    public Icon getIcon() {
+        return DefaultLoader.getUIHelper().loadIconByAction(AzureActionEnum.CREATE);
+    }
+
+    @Override
+    public void actionPerformed(NodeActionEvent e) {
+        final Project project = (Project) model.getProject();
+        try {
+            if (!AzureSignInAction.doSignIn(AuthMethodManager.getInstance(), project) ||
+                !AzureLoginHelper.isAzureSubsAvailableOrReportError(message("common.error.signIn"))) {
+                return;
+            }
+        } catch (final Exception ex) {
+            AzurePlugin.log(message("common.error.signIn"), ex);
+            DefaultLoader.getUIHelper().showException(message("common.error.signIn"), ex, message("common.error.signIn"), false, true);
+        }
+        final MySQLCreationDialog dialog = new MySQLCreationDialog(project);
+        dialog.setOkActionListener((data) -> this.createAzureMySQL(data, () -> DefaultLoader.getIdeHelper().invokeLater(dialog::close), project, dialog));
+        dialog.show();
+    }
+
+    private void createAzureMySQL(final AzureMySQLConfig config, Runnable callback, final Project project, MySQLCreationDialog dialog) {
```

Comment:
Why pass `DefaultLoader.getIdeHelper().invokeLater(dialog::close)` as a parameter instead of call it directly? Do we have any other value for `callback`?